src/xhity.c

```diff
@@ -745,9 +744,7 @@ int tary;
 							u.uy == tary - dy
 							)
 						{
-							struct monst *mdef2 = u.uswallow ? u.ustuck : 
```

Comment:
There's a regression here. This specific instance won't lead to the player attacking themselves, but it's unlikely they ever want to attack their own square.